robyn/openapi.py

```diff
@@ -0,0 +1,264 @@
+        :param query_params: query params for the function
+
+        :return: a tuple containing the endpoint with path params wrapped in braces and the "path" openapi object
+        according to spec
+        """
+        # robyn has paths like /:url/:etc whereas openapi requires path like /{url}/{path}
+        # this function is used for converting path params to the required form
+        # initialized with endpoint for handling endpoints without path params
+        endpoint_with_path_params_wrapped_in_braces = endpoint
+
+        endpoint_path_params_split = endpoint.split(":")
+        openapi_parameter_object = []
+
+        if len(endpoint_path_params_split) > 1:
+            endpoint_without_path_params = endpoint_path_params_split[0]
+
+            endpoint_with_path_params_wrapped_in_braces = (
+                endpoint_without_path_params[:-1] if endpoint_without_path_params.endswith("/") else endpoint_without_path_params
+            )
+
+            for path_param in endpoint_path_params_split[1:]:
+                path_param_name = path_param[:-1] if path_param.endswith("/") else path_param
+
+                openapi_parameter_object.append(
+                    {
+                        "name": path_param_name,
+                        "in": "path",
+                        "required": True,
+                        "schema": {"type": "string"},
+                    }
+                )
+                endpoint_with_path_params_wrapped_in_braces += "/{" + path_param_name + "}"
+
+        if query_params:
+            for query_param in query_params.__annotations__:
+                # ugly hack!
+                # returns "<class 'int'>" -- this line strips out the type a.k.a int from it
+                query_param_type = re.findall("'[a-z]+'", str(query_params.__annotations__[query_param]))[0].replace("'", "")
+
+                openapi_parameter_object.append(
+                    {
+                        "name": query_param,
+                        "in": "query",
+                        "required": False,
+                        "schema": {"type": query_param_type},
+                    }
+                )
+
+        return endpoint_with_path_params_wrapped_in_braces, {
+            "summary": summary,
+            "tags": tags,
+            "parameters": openapi_parameter_object,
+            "responses": {
+                "200": {
+                    "description": "Successful Response",
+                }
+            },
+        }
+
+    def docs_handler(self) -> FileResponse:
+        """
+        Handler to the swagger html page to be deployed to the endpoint `/docs`
+        @return: the swagger html page
+        """
+        html_file = str(Path("./robyn/swagger.html"))
+        return serve_html(html_file)
+
+    def json_handler(self) -> str:
+        """
+        Handler to the openapi spec json object to be deployed to the endpoint `/openapi.json`
```

Comment:
`json_handler` would usually return a json object. we should name it something else.